omg/core/views.py

```diff
@@ -0,0 +1,250 @@
+
+        Callback logic: 1 - Some of the pressed keys might not have their
+        corresponding `key_release` logic called when the view has been changed.
+        Those callbacks are called before the view is changed.
+        """
+        for (key_tuple, is_active) in self.active_keys.items():
+            # NOTE: keys which are also modifiers are pressed and released
+            # differently. CTRL is (65507, 18) when pressed, (65507, 16) when
+            # released. A more detailed check for those keys might be necessary
+            # in future if those are actively used.
+            if is_active:
+                self.on_key_release(*key_tuple)
+        self.active_keys = {}  # reset the dictionary
+
+    def _draw_ui(self):
+        for i, icon in enumerate(self.skill_icons):
+            x = self.icon_margin + i * (self.icon_size + self.icon_margin)
+            y = self.icon_margin
+            if i == self.player.skills.get_current_index():
+                arcade.draw_rectangle_outline(
+                    x + self.icon_size // 2,
+                    y + self.icon_size // 2,
+                    self.icon_size,
+                    self.icon_size,
+                    arcade.color.RED,
+                    border_width=3,
+                )
+            arcade.draw_texture_rectangle(
+                x + self.icon_size // 2,
+                y + self.icon_size // 2,
+                self.icon_size,
+                self.icon_size,
+                icon,
+            )
+
+
+class PauseView(arcade.View):
+    """Pause screen of the game."""
+
+    def __init__(self, window: arcade.Window):
+        super().__init__(window)
+        self._view_to_draw: arcade.View = None
+
+    def update_view_to_draw(self, new_view: arcade.View):
+        """Update the View to be drawn in the pause state."""
+        self._view_to_draw = new_view
+
+    def on_draw(self):
+        """Draw the game as is and the pause view extras."""
+        self.clear()
+        if self._view_to_draw:
+            self._view_to_draw.on_draw()
+
+        # Draw the Pause text on the given View as overlay
+        arcade.draw_text("PAUSED", SCREEN_WIDTH / 2, SCREEN_HEIGHT / 2 + 50,
+                         arcade.color.WHITE, font_size=50, anchor_x="center")
+
+        # Show tip to return or reset
+        arcade.draw_text("Press Esc. to return",
+                         SCREEN_WIDTH / 2,
+                         SCREEN_HEIGHT / 2,
+                         arcade.color.WHITE,
+                         font_size=20,
+                         anchor_x="center")
+        arcade.draw_text("Press Q to quit",
+                         SCREEN_WIDTH / 2,
+                         SCREEN_HEIGHT / 2 - 30,
+                         arcade.color.WHITE,
+                         font_size=20,
+                         anchor_x="center")
+
+    def on_key_release(self, key, modifiers):
+        """Key release logic."""
+        if key == arcade.key.Q:
```

Comment:
Can you please change the key here to another one since "Q" is currently used for iterating through collected elements?

Reply:
These are independent views. When this PauseView is active, the other one is disabled. So I think we should not think about other views when designing one. For me, "Q for Quit" is intuitive for now so that's why I put it.

In future, we might want to have a "EXIT" button where user needs to navigate and click to exit ( which is the case in most games). But I left the GUI design for later.

If you think Q is not an appropriate button, let's discuss it.